pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/metrics/impl/xml/AtfdTest.xml

```diff
@@ -125,16 +125,17 @@ public class StatementAndBraceFinder extends JavaParserVisitorAdapter {
 
 
     <test-code>
-        <description>Full example</description> <!-- TODO issues w/ that-->
-        <expected-problems>7</expected-problems>
+        <description>Full example</description>
+        <expected-problems>8</expected-problems>
         <expected-messages>
-            <message>'StatementAndBraceFinder' has value 36 highest 18.</message>
-            <message>'StatementAndBraceFinder#buildDataFlowFor(JavaNode)' has value 6.</message>
-            <message>'StatementAndBraceFinder#tryToLog(String, NodeType, Node)' has value 4.</message>
+            <message>'StatementAndBraceFinder' has value 18.</message>
+            <message>'StatementAndBraceFinder#StatementAndBraceFinder(DataFlowHandler)' has value 0.</message>
+            <message>'StatementAndBraceFinder#buildDataFlowFor(JavaNode)' has value 4.</message>
+            <message>'StatementAndBraceFinder#tryToLog(String, NodeType, Node)' has value 2.</message>
             <message>'StatementAndBraceFinder#tryToLog(NodeType, Node)' has value 0.</message>
-            <message>'StatementAndBraceFinder#visit(ASTStatementExpression, Object)' has value 4.</message>
-            <message>'StatementAndBraceFinder#visit(ASTVariableDeclarator, Object)' has value 4.</message>
-            <message>'StatementAndBraceFinder#visit(ASTExpression, Object)' has value 18.</message>
+            <message>'StatementAndBraceFinder#visit(ASTStatementExpression, Object)' has value 2.</message>
+            <message>'StatementAndBraceFinder#visit(ASTVariableDeclarator, Object)' has value 2.</message>
+            <message>'StatementAndBraceFinder#visit(ASTExpression, Object)' has value 6.</message>
```

Comment:
I double checked atfd and I think they are. I simplified the test code and refined the metric's definition a bit.

Reply:
More details: the method which shows complexity 18 has 6 times the following kind of block:
```
            dataFlow.createNewNode(node); // FOR EXPR
            dataFlow.pushOnStack(NodeType.FOR_EXPR, dataFlow.getLast());
            tryToLog(loggerTag, NodeType.FOR_EXPR, node);
```

Previously this was added up to 3, because there are two field accesses to NodeType.FOR_EXPR + one getter call. Now we only count the getter call as accessing a constant doesn't look problematic. So instead of ATFD=6x3 we have =6x1.

The other methods have higher counts because they count `LOGGER.isLoggable(Level.FINEST)` as +2 (+1 for the field access and +1 for the method call). We don't count the field because it's static, nor the method, because it has a parameter and hence is not a real getter.